### DISCUSSION TITLE
Github actions nuget binary cache fails to upload

### DISCUSSION THREAD

**Discussion by mathisloge** [microsoft/vcpkg#20804] [Q&A] [answered]
So i've set up a github action as in [CI Examples - GitHub Packages](https://github.com/microsoft/vcpkg/blob/master/docs/users/binarycaching.md#github-packages). However for most packages I'm getting `Pushing NuGet to GitHub failed. Use --debug for more information.`. With --debug enabled, this is what I get for most of those packages:
```
Uploading binaries for boost-preprocessor:x64-linux to NuGet source GitHub.
[DEBUG] popen(/usr/bin/mono /home/runner/work/mapnik/mapnik/vcpkg/downloads/tools/nuget-5.10.0-linux/nuget.exe push /home/runner/work/mapnik/mapnik/vcpkg/buildtrees/boost-preprocessor_x64-linux.1.77.0-vcpkg1bf373b633de6d09d309f88af09ca48fdd6015a5ce3f97bf19d410942a9cf387.nupkg -ForceEnglishOutput -Timeout 100 -Source GitHub -NonInteractive 2>&1)
[DEBUG] cmd_execute_and_stream_data() returned 256 after  1596026 us
WARNING: No API Key was provided and no API Key could be found for 'https://nuget.pkg.github.com/mathisloge'. To save an API Key for a source use the 'setApiKey' command.
Pushing boost-preprocessor_x64-linux.1.77.0-vcpkg1bf373b633de6d09d309f88af09ca48fdd6015a5ce3f97bf19d410942a9cf387.nupkg to 'https://nuget.pkg.github.com/mathisloge'...
  PUT https://nuget.pkg.github.com/mathisloge/
WARNING: Package "boost-preprocessor_x64-linux" is already associated with another repository.
  UnprocessableEntity https://nuget.pkg.github.com/mathisloge/ 767ms
Response status code does not indicate success: 422 (Unprocessable Entity).
```

The root cause is probably the naming of the packages `Package "boost-preprocessor_x64-linux" is already associated with another repository`. Is there a way to rename the resulting packages e.g. `boost-preprocessor_x64-linux_my_project`? 

The github run: https://github.com/mathisloge/mapnik/runs/3919777415

**Comment by Neumann-A** [comment, marked as answer]
The trick with github packages is to have one repo for packages only.

**Reply by mathisloge** [reply]
So instead of `-source "https://nuget.pkg.github.com/mathisloge/index.json"` i would use `-source "https://nuget.pkg.github.com/mathisloge/mapnik/index.json"`?

Or should I use the same vcpkg version across all projects?

**Reply by Neumann-A** [reply]
keep everything the same but set env variable `VCPKG_NUGET_REPOSITORY`

**Reply by mathisloge** [reply]
Which values do I have to set? (at xxx) 

```
env:
  VCPKG_NUGET_REPOSITORY: "xxx"
```
I've tried to add a newly created repo to be used for all other repos to use. But I don't really know what I have to set. 
e.g. `https://github.com/mathisloge/vcpkg-nuget.git` or `vcpkg-nuget` or `mathisloge/vcpkg-nuget` are producing the following message: 

```
Uploading binaries for boost-config:x64-linux to NuGet source GitHub.
[DEBUG] popen(/usr/bin/mono /usr/local/share/vcpkg/downloads/tools/nuget-5.10.0-linux/nuget.exe push /usr/local/share/vcpkg/buildtrees/boost-config_x64-linux.1.77.0-vcpkg714b86d7a6af2913acf4f30ae6e64b3797ec6ff2b93e75e86ed68478cee7c61a.nupkg -ForceEnglishOutput -Timeout 100 -Source GitHub -NonInteractive 2>&1)
[DEBUG] cmd_execute_and_stream_data() returned 256 after  1516674 us
WARNING: No API Key was provided and no API Key could be found for 'https://nuget.pkg.github.com/mathisloge'. To save an API Key for a source use the 'setApiKey' command.
Pushing boost-config_x64-linux.1.77.0-vcpkg714b86d7a6af2913acf4f30ae6e64b3797ec6ff2b93e75e86ed68478cee7c61a.nupkg to 'https://nuget.pkg.github.com/mathisloge'...
  PUT https://nuget.pkg.github.com/mathisloge/
WARNING: Resource not accessible by integration
MSBuild auto-detection: using msbuild version '15.0' from '/usr/lib/mono/msbuild/15.0/bin'.
Using credentials from config. UserName: mathisloge
WARNING: Resource not accessible by integration
  Forbidden https://nuget.pkg.github.com/mathisloge/ 723ms
Response status code does not indicate success: 403 (Forbidden).
Pushing NuGet to GitHub failed. Use --debug for more information.
```

**Reply by Neumann-A** [reply]
Feel free to check https://github.com/Neumann-A/StochasticPhysics/blob/dev/.github/workflows/ci.windows.yml
`VCPKG_NUGET_REPOSITORY` is set to the git repo.
There is a step called "Setup NuGet credentials" and a step called "Setup NuGet apikey"  (https://github.com/Neumann-A/StochasticPhysics/blob/442b856580d4f25fe568ce68a0e111e1576419f0/.github/workflows/ci.windows.yml#L90-L94) after that

**Reply by mathisloge** [reply]
Ah perfect thanks! 
Didn't noticed that the GITHUB_TOKEN wasn't able to access cross repo.
